fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/ContentObject.java

```diff
@@ -31,11 +40,79 @@ protected ContentObject(PID pid, Repository repository, RepositoryObjectDataLoad
 		super(pid, repository, dataLoader);
 	}
 
-	public BinaryObject addDescription(InputStream contentStream) {
-		return null;
+	/**
+	 * Adds MODS to the object by creating the necessary relations in Fedora
+	 * @param modsStream
+	 * @return the FileObject containing the BinaryObject for the MODS
+	 */
+	public FileObject addDescription(InputStream modsStream) {
+		FileObject fileObj = createFileObject();
+		
+		BinaryObject mods = fileObj.addOriginalFile(modsStream, null, "text/xml", null);
+		repository.createRelationship(pid, PcdmModels.hasRelatedObject, fileObj.getResource());
+		repository.createRelationship(pid, Cdr.hasMods, mods.getResource());
+		return fileObj;
+	}
+	
+	/**
+	 * Adds source metadata and MODS to the object by creating the necessary relations in Fedora
+	 * @param sourceMdStream
+	 * @param sourceProfile
+	 * @param modsStream
+	 * @return a FileObject containing BinaryObejcts for source metadata and MODS
```

Comment:
BinaryObjects

Also, for the description of this method, could you make it something like "Adds description information to this object, containing a source metadata and a MODS record derived from it."  

It may be helpful to also state what the sourceProfile is, so something like "identifies the encoding, profile, and/or origins of the sourceMdStream using an identifier defined in edu.unc.lib.dl.util.MetadataProfileConstants".  I just figure the details of how source metadata could be confusing to future generations.